tests/test_embedding_client_local.py

```
@@ -0,0 +1,137 @@
+    monkeypatch.setattr("src.embedding_client.settings.LLM.EMBEDDING_PROVIDER", "local")
+    monkeypatch.setattr("src.embedding_client.settings.LLM.EMBEDDING_BASE_URL", None)
+
+    with pytest.raises(ValueError, match="LLM_EMBEDDING_BASE_URL is required"):
+        _EmbeddingClient(provider="local")
+
+
+def test_openrouter_prefers_embedding_overrides(monkeypatch: pytest.MonkeyPatch):
+    created: dict = {}
+
+    class _FakeEmbeddingsAPI:
+        async def create(self, *, model: str, input: str | list[str]):
```

Ruff (0.15.9)

[error] Function argument `input` is shadowing a Python builtin

(A002)